privacyidea/api/lib/utils.py

```diff
@@ -61,6 +61,57 @@
 required = False
 
 
+def _check_allowed_param(ret, key, default, allow_empty, allowed_values):
+    if not allow_empty and ret == "":
+        raise ParameterError(f"Parameter {key} must not be empty", id=905)
+    if allowed_values and ret not in allowed_values:
+        ret = default
+    return ret
+
+
+def _get_param(param, key, default=None):
+    ret = None
+    if param and key in param:
+        ret = param[key]
+    elif default:
+        ret = default
+    return ret
+
+
+def get_required(param, key, default=None, allow_empty=False, allowed_values=None):
+    ret = _get_param(param, key, default)
+    if ret is None:
+        raise ParameterError(f"Missing parameter: {key}", id=905)
+    return _check_allowed_param(ret, key, default, allow_empty, allowed_values)
+
+
+def get_required_one_of(param, keys, default=None, allow_empty=False, allowed_values=None):
```

Comment:
remove stupid input params: why allow_empty if the function is called get_required